walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,108 @@
 
 package com.google.sps;
 
+import java.util.Collections;
 import java.util.Collection;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Arrays; 
+
 
 public final class FindMeetingQuery {
+
+  private final int MINUTES_IN_A_DAY = 1440;
+
+  /**
+   * Returns an {@ArrayList} of {@code TimeRange}s of when an Event can take place.
+   * Runtime: O(1) -> Regardless of the amount of events/attendees, a day will always
+   * have the same number of minutes, which is what the array represents.
+   * @param busyTimes boolean array containing all times where there is a meeting.
+   * @param request Desired duration of the requested event
+   * @return {@code ArrayList} with all possible timeranges for an event to happen.
+   */
+  private ArrayList<TimeRange> findTimeRanges(boolean[] busyTimes, long duration){
+    ArrayList<TimeRange> possibleTimes = new ArrayList<TimeRange>();
+    // Find free time in the array and if they fit the meeting, create the time range
+    int availabilityInterval = 0;
+    for(int minuteOfTheDay = 0; minuteOfTheDay < MINUTES_IN_A_DAY+1; minuteOfTheDay++){
+      // If the minute is not busy, add a free minute to the counter.
+      if(!busyTimes[minuteOfTheDay]){
+        availabilityInterval++; 
+      }else if(availabilityInterval != 0){
+        if(availabilityInterval >= duration){
+          TimeRange tr = TimeRange.fromStartDuration(minuteOfTheDay-availabilityInterval, availabilityInterval); 
+          possibleTimes.add(tr);
+        }
+        availabilityInterval = 0;
+      }
+    }
+
+    // If the whole day is available return a single TimeRange
+    if(availabilityInterval == MINUTES_IN_A_DAY+1){
+      possibleTimes.add(TimeRange.WHOLE_DAY);
+      return possibleTimes;
+    }
+
+    // In case there is no event later, this will add the time since the last event until EoD.
+    if(busyTimes.length - availabilityInterval  >= duration && availabilityInterval != 1){
```

Comment:
Wouldn't availabilityInterval == 1 if the last minute of the day is available? Shouldn't that be a possibleTime if my requested meeting duration is only 1 minute?

Reply:
Yeah but the array is created with an extra "slot" so technically that 1 is accounting for that. When that `availabilityInterval == 1` isn't there a TimeRange of [1440,1440) is created. So it's a meeting that has a duration of 0

Reply:
I went ahead and added a test for this case. It contains a meeting that lasts from START_OF_DAY all the way to END_OF_DAY (non inclusive of 11:59PM), and the test checks if query returns a single meeting from 1439 to 1440, which is the EoD

Reply:
Cool, sounds good. Thanks for explaining and adding the test case!